stellargraph/utils/history.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import matplotlib.pyplot as plt
+
+
+__all__ = ["plot_history"]
+
+
+def plot_history(
+    history,
+    individual_figsize=(7, 4),
+    color_train=None,
+    color_validation=None,
+    **kwargs
+):
+    """
+    Plot the training history of one or more models.
+
+    This creates a column of plots, with one plot for each metric recorded during training, with the
+    plot showing the metric vs. epoch. If multiple models have been trained (that is, a list of
+    histories is passed in), each metric plot includes multiple train and validation series.
+
+    Validation data is optional (it is detected by metrics with names starting with ``val_``).
+
+    Args:
+        history: the training history, as returned by :meth:`tf.keras.Model.fit`
+        individual_figsize (tuple of numbers): the size of the plot for each metric
+        color_train: a matplotlib color to use for the train series for each metric
+        color_validation: a matplotlib color to use for the validation series for each metric
+        kwargs: additional arguments to pass to :meth:`matplotlib.pyplot.subplots`
+    """
+
+    # explicit colours are needed if there's multiple train or multiple validation series, because
+    # each train series should have the same color. This uses the global matplotlib defaults that
+    # would be used for a single train and validation series.
+    colors = plt.rcParams["axes.prop_cycle"].by_key()["color"]
+    if color_train is None:
+        color_train = colors[0]
+    if color_validation is None:
+        color_validation = colors[1]
```

Comment:
Should we make sure that the two colours aren't the same? It is pretty unlikely.. but technically if someone wanted a particular colour for `train` but didn't care about the validation colour, it might be kind of convenient.

Reply:
Oh, like if they set `color_train="blue"` and `color_validation`'s default is `"blue"` too. That's a good point.

I think I might just remove the colour customisation entirely for now: the default colours are good enough for our examples (e.g. they seem distinguishable for all colourblindness types supported by https://www.color-blindness.com/coblis-color-blindness-simulator/), and I suspect someone wanting different colours is likely to want other customisations too, like changing axis names and fonts and ....